src/openrct2/drawing/Sprite.cpp

```diff
@@ -36,6 +36,89 @@
 using namespace OpenRCT2;
 using namespace OpenRCT2::Ui;
 
+static inline uint32 rctc_to_rct2_index(uint32 image)
+{
+    if      (                  image <  1575) return image;
+    else if (image >=  1607 && image <  4983) return image - 32;
+    else if (image >=  4986 && image < 17189) return image - 35;
+    else if (image >= 17191 && image < 18121) return image - 37;
+    else if (image >= 18123 && image < 23800) return image - 39;
+    else if (image >= 23804 && image < 24670) return image - 43;
+    else if (image >= 24674 && image < 28244) return image - 47;
+    else if (image >= 28246                 ) return image - 49;
+    else throw Exception("Invalid RCTC g1.dat file");
```

Comment:
Can you really infer it's an issue with g1.dat just from the requested image id?
Where does this exception end up, how is it presented to the user? What options does the user have, can he ignore it and continue with invalid g1.dat?
By the looks of it, it's checked upon loading g1.dat, right?

Reply:
It will result in the generic message that the user has selected the wrong directory for RCT2 files.

Reply:
>Can you really infer it's an issue with g1.dat just from the requested image id?

We know it's an issue with g1.dat because of how the function is being used. This function is only called when figuring out `zoomed_offset` of the element (as we have to account for the elements being skipped). The only way that `image` is one of the 49 values that cause the exception is if the element being processed from g1.dat sets its `zoomed_offset` to point at one of the elements we skip (which none of the g1.dat files from any of the releases I tested of RCTC, PC iOS and Android, will ever have that happen).

>Where does this exception end up, how is it presented to the user? What options does the user have, can he ignore it and continue with invalid g1.dat?

Right now, the exception gets caught by `gfx_load_g1`, where it produces a message to the user telling them there's a problem with the RCT2 directory they selected, and return false up the call chain. The false return code appears to be (improperly) ignored a ways up, so the game will attempt to play without any of the sprites, and would likely crash shortly after. However, that's a separate issue to what's being added here.

As to whether this is an assert or exception is probably more of a style thing than anything. Personally, I think using `assert` would be a mistake, as `assert` should be reserved for checking for programming errors, and this handles a file parsing error. But I need to get to sleep now, so feel free to decide what to do with this, and let me know what you decide.